### DISCUSSION TITLE
Import

### DISCUSSION THREAD

**Discussion by jgomez720** [KittyCAD/modeling-app#478] [New Feature Requests]
### Description
A tool that enables the user to import their own CAD files into their work environment.

### Functionality
1. User selects "import" button (I've seen this found in multiple different areas. File -> import, right click -> import, OnShape's plus sign "+" -> import from CAD). No specific place to put it, just as long as it's intuitive to the user. Perhaps we could even have a drag-and-drop feature.
2. Depending on the CAD file, there may be some options that the user has to select. For example, some CAD packages will ask how they want their file oriented in the workspace. 
3. Depending on the type of file imported, users will have different levels of interactivity. STEP file contains BREP info, so users can typically still use faces and edges. OBJs are meshes, and usually, you can only interact with the body as a whole. Even if a face is flat, the user still is unable to click on it.

**Comment by jgomez720** [comment]
Supported import formats would align with api docs. 
* GLTF
* OBJ
* PLY
* STEP
* SLDPRT
* STL

**Reply by jgomez720** [reply]
Eventually, NX as well (.prt)

**Comment by bitjeep** [comment]
Thought I was completely daft because I couldn't figure out how to bring a 3D file in. Finally realized it hadn't been implemented yet. Might just want to aim for making it view-only on the first cut since, as you said in point 3, editing all the different types of 3D files will likely be complicated.

**Comment by jessfraz** [comment]
added some known issues but closing, the known issues will take care of edit and transform and move in the future

**Comment by alteous** [comment]
Is there an option for import now in the modelling app? I haven't found it yet on app.zoo.dev.
